addons/base/views.py

```diff
@@ -302,40 +302,29 @@ def get_authenticated_resource(resource_id):
     return resource
 
 
-def get_file_version_from_wb(waterbutler_data: dict) -> FileVersion:
-    path = waterbutler_data.get('path')
-    if waterbutler_data['provider'] != 'osfstorage' or not path:
-        return
-
-    path = waterbutler_data.get('path')
-    file_id = path.strip('/')
-    version = waterbutler_data.get('version')
-    if version:
-        try:
-            return FileVersion.objects.filter(
-                basefilenode___id=file_id,
-                identifier=int(version)
-            ).select_related('region').get()
-        except FileVersion.DoesNotExist:
-            raise HTTPError(http_status.HTTP_400_BAD_REQUEST)
-    else:
-        try:
-            return BaseFileNode.active.get(
-                _id=file_id,
-            ).versions.order_by(
-                '-created'
-            ).first()
-        except BaseFileNode.DoesNotExist:
-            raise HTTPError(http_status.HTTP_400_BAD_REQUEST)
+def _get_osfstorage_file_version(file_node: OsfStorageFileNode, version_string: str = None) -> FileVersion:
+    if not (file_node and file_node.is_file):
+        return None
+
+    version = int(version_string) if version_string else file_node.versions.count()
+    try:
+        return FileVersion.objects.select_related('region').get(
+            basefilenode=file_node,
```

Comment:
Historically, this was grabbed by `basefilenode___id=file_id` even though the filenode had already been loaded. I'm hoping that this was just an oversight:
https://github.com/CenterForOpenScience/osf.io/blob/b2916f756c792bb9eb8f3252e0b6fc56c5b0618c/addons/base/views.py#L321

Reply:
Likely oversight. It should be slightly more efficient to use foreign relations like this, rather than a `JOIN` querying on a text field.